homeassistant/components/switch/light.py

```diff
@@ -59,6 +60,27 @@ async def async_setup_platform(
     )
 
 
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Initialize Light Switch config entry."""
+
+    registry = er.async_get(hass)
+    entity_id = er.async_resolve_entity(registry, config_entry.options[CONF_ENTITY_ID])
```

Comment:
This can be `None` if entity it is based on is deleted. We should detect this inside `__init__.py` async_setup_entry and raise ConfigEntryNotReady.

Reply:
I like checking this early, but raising `ConfigEntryNotReady` will keep retrying which seems pointless:
```
homeassistant.config_entries:config_entries.py:366 Config entry 'ABC' for switch integration not ready yet: Unknown entity registry entry a266a680b608c32770e6c45bfe6b8411; Retrying in background
```

We should rather raise anything *but* `ConfigEntryNotReady` to have the config entry setup fail?